.github/workflows/linux.yml

```diff
@@ -132,99 +119,157 @@ jobs:
+        with:
+          name: openvino_package
+          path: ${{ env.BUILD_DIR }}/openvino_package.tar.gz
+          if-no-files-found: 'error'
 
-      - name: Install python wheels
-        run: python3 -m pip install openvino-dev --find-links=${INSTALL_DIR}/tools
+      - name: Upload openvino tests package
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: openvino_tests
+          path: ${{ env.BUILD_DIR }}/openvino_tests.tar.gz
+          if-no-files-found: 'error'
 
-      - name: Install tests
-        run: cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DCOMPONENT=tests -P ${BUILD_DIR}/cmake_install.cmake
+  Samples:
+    needs: Build
+    defaults:
+      run:
+        shell: bash
+    runs-on: ubuntu-20.04
+    container:
+      image: ubuntu:20.04
+    env:
+      DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
+      INSTALL_DIR: /__w/openvino/openvino/install
+      INSTALL_TEST_DIR: /__w/openvino/openvino/install/tests
+      BUILD_DIR: /__w/openvino/openvino/build
 
-      - name: Install OpenVINO
-        run: cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -P ${BUILD_DIR}/cmake_install.cmake
+    steps:
+      - name: Create Directories
+        run: mkdir -p ${INSTALL_DIR} ${INSTALL_TEST_DIR}
 
-      - name: CMake Samples Tests
-        run: cmake -GNinja -S ${OPENVINO_REPO}/tests/samples_tests -B ${BUILD_DIR}/samples_tests
+      #
+      # Initialize OpenVINO
+      #
 
-      - name: Build Samples Tests
-        run: cmake --build ${BUILD_DIR}/samples_tests --config Release
+      - name: Download OpenVINO package
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_package
+          path: ${{ env.INSTALL_DIR }}
 
-      - name: Install Samples Tests
-        run: cmake -DCOMPONENT=tests -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -P ${BUILD_DIR}/samples_tests/cmake_install.cmake
+      - name: Download OpenVINO tests package
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_tests
+          path: ${{ env.INSTALL_TEST_DIR }}
 
-      - name: Pack Artifacts
+      - name: Extract OpenVINO packages
         run: |
           pushd ${INSTALL_DIR}
-            tar -czvf ${BUILD_DIR}/openvino_package.tar.gz --exclude=tests *
+            tar -xzf openvino_package.tar.gz -C ${INSTALL_DIR} && rm openvino_package.tar.gz || exit 1
           popd
-
-          pushd ${INSTALL_DIR}
-            tar -czvf ${BUILD_DIR}/openvino_tests.tar.gz tests/
+          pushd ${INSTALL_TEST_DIR}
+            tar -xzf openvino_tests.tar.gz -C ${INSTALL_DIR} && rm openvino_tests.tar.gz || exit 1
           popd
 
-      - name: Build cpp samples
-        run: ${SAMPLES_INSTALL_DIR}/cpp/build_samples.sh -i ${INSTALL_DIR} -b ${BUILD_DIR}/cpp_samples
+      - name: Install 'actions/setup-python@v4' dependencies
+        run: apt-get update && apt-get install -y libssl1.1
+
+      - uses: actions/setup-python@v4
```

Comment:
do we really need ti set python env for those steps?

Reply:
yes, because samples tests are written in python.

Reply:
yes, but we need it only for the tests, it would be better to move down this action to the Tests section